Build failure: kmymoney

### Nixpkgs version

- Unstable (25.11)
- `master` (tested at 35b2fb28ac764b48bd07225ccb89ece03f468c00)

### Steps to reproduce

1. `nix build github:nixos/nixpkgs#kmymoney`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310591811

### Relevant log output

```console
[ 14%] Building CXX object kmymoney/plugins/sqlcipher/CMakeFiles/qsqlcipher.dir/
[…]
/build/kmymoney-5.2.1/kmymoney/plugins/sqlcipher/qsql_sqlite.cpp:37:10: fatal error: QtSql/private/qsqlcachedresult_p.h: No such file or directory
   37 | #include <QtSql/private/qsqlcachedresult_p.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [kmymoney/plugins/sqlcipher/CMakeFiles/qsqlcipher.dir/build.make:113: kmymoney/plugins/sqlcipher/CMakeFiles/qsqlcipher.dir/qsql_sqlite.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7043: kmymoney/plugins/sqlcipher/CMakeFiles/qsqlcipher.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[…]
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/jc3mfa7ybzh32bkvdj5xiib1vkx8jy6x-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.49, NixOS, 25.11 (Xantusia), 25.11.20250928.e9f00bd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/das-g/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/das-g/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/das-g/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/das-g/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/das-g/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/h3kk1353iq1msxr8s3lha1iac7icsqy2-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/23a2fc0ha64j2ygaq8b9gaqc5sb0whf7-lix-2.93.3/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/jc3mfa7ybzh32bkvdj5xiib1vkx8jy6x-source`

### Notify maintainers

* @das-g (That's me. 😅)

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/km/kmymoney/kmymoney-fix-build-against-qt-6-10.patch
+++ b/pkgs/by-name/km/kmymoney/kmymoney-fix-build-against-qt-6-10.patch
@@ -1,0 +1,30 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 32e2840..f21e7ed 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -183,7 +183,7 @@ if (PkgConfig_FOUND)
+ endif()
+ 
+ find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} REQUIRED COMPONENTS Core DBus Widgets Svg Xml Test PrintSupport)
+-find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} OPTIONAL_COMPONENTS Sql Concurrent QuickWidgets)
++find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} OPTIONAL_COMPONENTS Sql SqlPrivate Concurrent QuickWidgets)
+ 
+ find_package(KF${QT_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS Archive CoreAddons Config ConfigWidgets Crash I18n Completion KCMUtils ItemModels ItemViews Service XmlGui TextWidgets Notifications KIO)
+ 
+diff --git a/kmymoney/plugins/sqlcipher/CMakeLists.txt b/kmymoney/plugins/sqlcipher/CMakeLists.txt
+index ee6f2e7..aa3fc19 100644
+--- a/kmymoney/plugins/sqlcipher/CMakeLists.txt
++++ b/kmymoney/plugins/sqlcipher/CMakeLists.txt
+@@ -32,10 +32,8 @@ target_link_libraries(qsqlcipher
+   PRIVATE
+     PkgConfig::SQLCIPHER
+ )
+-if (TARGET Qt::SqlPrivate)
+-  target_link_libraries(qsqlcipher PRIVATE Qt::SqlPrivate)
+-endif()
+-target_link_libraries(qsqlcipher PRIVATE Qt::Sql)
++
++target_link_libraries(qsqlcipher PRIVATE Qt::Sql Qt::SqlPrivate)
+ 
+ if(BUILD_TESTING)
+  add_subdirectory(tests)

--- a/pkgs/by-name/km/kmymoney/package.nix
+++ b/pkgs/by-name/km/kmymoney/package.nix
@@ -77,6 +77,11 @@ stdenv.mkDerivation (finalAttrs: {
     python3.pkgs.woob
   ];
 
+  patches = [
+    # from https://src.fedoraproject.org/rpms/kmymoney/c/8f7f40d7fec6db96610e60a6a99717479594c8bd
+    ./kmymoney-fix-build-against-qt-6-10.patch
+  ];
+
   postPatch = ''
     buildPythonPath "${python3.pkgs.woob}"
     patchPythonScript "kmymoney/plugins/woob/interface/kmymoneywoob.py"

